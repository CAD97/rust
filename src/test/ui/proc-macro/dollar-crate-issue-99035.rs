// check-pass
// edition:2018
// compile-flags: -Z span-debug
// aux-build:test-macros.rs
// aux-build:dollar-dollar-crate-external.rs

#![no_std] // Don't load unnecessary hygiene information from std
extern crate std;

#[macro_use]
extern crate test_macros;
extern crate dollar_dollar_crate_external;

struct S;

mod namespace {
    use dollar_dollar_crate_external::define_macro;

    define_macro!(m => S);

    #[print_attr]
    const _: crate::S = m!();
}

fn main() {}
