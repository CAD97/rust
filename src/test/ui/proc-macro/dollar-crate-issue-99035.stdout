PRINT-ATTR INPUT (DISPLAY): const _ : crate :: S = m! () ;
PRINT-ATTR INPUT (DEBUG): TokenStream [
    Ident {
        ident: "const",
        span: $DIR/dollar-crate-issue-99035.rs:22:5: 22:10 (#0),
    },
    Ident {
        ident: "_",
        span: $DIR/dollar-crate-issue-99035.rs:22:11: 22:12 (#0),
    },
    Punct {
        ch: ':',
        spacing: Alone,
        span: $DIR/dollar-crate-issue-99035.rs:22:12: 22:13 (#0),
    },
    Ident {
        ident: "crate",
        span: $DIR/dollar-crate-issue-99035.rs:22:14: 22:19 (#0),
    },
    Punct {
        ch: ':',
        spacing: Joint,
        span: $DIR/dollar-crate-issue-99035.rs:22:19: 22:21 (#0),
    },
    Punct {
        ch: ':',
        spacing: Alone,
        span: $DIR/dollar-crate-issue-99035.rs:22:19: 22:21 (#0),
    },
    Ident {
        ident: "S",
        span: $DIR/dollar-crate-issue-99035.rs:22:21: 22:22 (#0),
    },
    Punct {
        ch: '=',
        spacing: Alone,
        span: $DIR/dollar-crate-issue-99035.rs:22:23: 22:24 (#0),
    },
    Ident {
        ident: "m",
        span: $DIR/dollar-crate-issue-99035.rs:22:25: 22:26 (#0),
    },
    Punct {
        ch: '!',
        spacing: Alone,
        span: $DIR/dollar-crate-issue-99035.rs:22:26: 22:27 (#0),
    },
    Group {
        delimiter: Parenthesis,
        stream: TokenStream [],
        span: $DIR/dollar-crate-issue-99035.rs:22:27: 22:29 (#0),
    },
    Punct {
        ch: ';',
        spacing: Alone,
        span: $DIR/dollar-crate-issue-99035.rs:22:29: 22:30 (#0),
    },
]
